cli/src/opts/cast.rs

```diff
@@ -137,6 +140,18 @@ pub enum Subcommands {
         #[structopt(flatten)]
         eth: EthereumOpts,
     },
+    #[structopt(name = "abi-decode")]
+    #[structopt(about = "Decode ABI-encoded hex data")]
+    AbiDecode {
+        #[structopt(
+            help = "the function signature you want to decode, in the format `<name>(<in-types>)(<out-types>)`"
+        )]
+        sig: String,
+        #[structopt(short = "c", help = "the encoded calladata, in hex format")]
```

Comment:
Make this a positional argument (remove the long/short). Also there's a typo in the doc "calldata"